NitroxClient/MonoBehaviours/NitroxEntity.cs

```diff
@@ -61,6 +94,27 @@ public static void SetNewId(GameObject gameObject, NitroxId id)
                 entity = gameObject.AddComponent<NitroxEntity>();
             }
 
+            List<GameObject> childGameObjects = new List<GameObject>();
+
+            if (gameObjectsWaitingForParent.TryGetValue(id, out childGameObjects))
+            {
+                foreach (GameObject childGameObject in childGameObjects)
+                {
+                    if (childGameObject == null)
+                    {
+                        continue; // Child GameObject was deleted by the game?
+                    }
+
+                    childGameObject.transform.SetParent(gameObject.transform, true);
+                    if (childGameObject.GetComponent<LargeWorldEntity>() != null)
+                    {
+                        LargeWorldEntity.Register(childGameObject);
+                    }
+                }
+
+                gameObjectsWaitingForParent.Remove(id);
+            }
+
```

Comment:
I don't think this logic belong here.  We do not need to do this look up every time we add an id - we add ids all the time to game objects.